From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Erfan Abdi <erfangplus@gmail.com>
Date: Tue, 18 May 2021 08:33:28 +0430
Subject: [PATCH] Add support for host hals whitelist

---
 Android.bp                      |  3 ++
 transport/ServiceManagement.cpp | 64 +++++++++++++++++++++++++++------
 2 files changed, 56 insertions(+), 11 deletions(-)

diff --git a/Android.bp b/Android.bp
index e0ca9e2..dba23bd 100644
--- a/Android.bp
+++ b/Android.bp
@@ -141,6 +141,9 @@ cc_defaults {
     static_libs: [
         "libhwbinder-impl-internal",
     ],
+    whole_static_libs: [
+        "libtinyxml2",
+    ],
 
     target: {
         android: {
diff --git a/transport/ServiceManagement.cpp b/transport/ServiceManagement.cpp
index 9884c70..6c17884 100644
--- a/transport/ServiceManagement.cpp
+++ b/transport/ServiceManagement.cpp
@@ -37,6 +37,7 @@
 #include <hidl/ServiceManagement.h>
 #include <hidl/Status.h>
 #include <utils/SystemClock.h>
+#include <tinyxml2.h>
 
 #include <android-base/file.h>
 #include <android-base/logging.h>
@@ -60,6 +61,9 @@ using IServiceManager1_1 = android::hidl::manager::V1_1::IServiceManager;
 using IServiceManager1_2 = android::hidl::manager::V1_2::IServiceManager;
 using ::android::hidl::manager::V1_0::IServiceNotification;
 
+using NodeType = tinyxml2::XMLElement;
+using DocType = tinyxml2::XMLDocument;
+
 namespace android {
 namespace hardware {
 
@@ -151,6 +155,24 @@ __attribute__((noinline)) static void tryShortenProcessName(const std::string& d
     }
 }
 
+static bool isHostHalAllowed(std::string ifaceName, bool* priority) {
+    DocType doc;
+    doc.LoadFile("/system/etc/hosthals.xml");
+    NodeType *titleElement = doc.FirstChildElement("hosthals");
+
+    for (NodeType *child = titleElement->FirstChildElement("hal");
+         child != nullptr;
+         child = child->NextSiblingElement("hal"))
+    {
+        if (!strcmp(ifaceName.c_str(), child->FirstChildElement("name")->GetText())) {
+            *priority = !strcmp("true", child->FirstChildElement("priority")->GetText());
+            return true;
+        }
+    }
+
+    return false;
+}
+
 namespace details {
 
 static void onRegistrationImpl(const std::string& descriptor, const std::string& instanceName) {
@@ -322,6 +344,7 @@ struct PassthroughServiceManager : IServiceManager1_1 {
         const std::string& fqName,
         const std::function<bool /* continue */ (void* /* handle */, const std::string& /* lib */,
                                                  const std::string& /* sym */)>& eachLib) {
+        bool hostHalPriority = false;
         //fqName looks like android.hardware.foo@1.0::IFoo
         size_t idx = fqName.find("::");
 
@@ -333,6 +356,7 @@ struct PassthroughServiceManager : IServiceManager1_1 {
 
         std::string packageAndVersion = fqName.substr(0, idx);
         std::string ifaceName = fqName.substr(idx + strlen("::"));
+        std::string ifaceNameNoVer = fqName.substr(0, fqName.find("@"));
 
         const std::string prefix = packageAndVersion + "-impl";
         const std::string sym = "HIDL_FETCH_" + ifaceName;
@@ -342,15 +366,27 @@ struct PassthroughServiceManager : IServiceManager1_1 {
 
         dlerror(); // clear
 
+        bool isHostHal = isHostHalAllowed(ifaceNameNoVer, &hostHalPriority);
         static std::string halLibPathVndkSp = android::base::StringPrintf(
             HAL_LIBRARY_PATH_VNDK_SP_FOR_VERSION, details::getVndkVersionStr().c_str());
-        std::vector<std::string> paths = {
-            HAL_LIBRARY_PATH_ODM, HAL_LIBRARY_PATH_VENDOR,
-            HAL_LIBRARY_PATH_VENDOR_EXTRA, halLibPathVndkSp,
+
+        std::vector<std::string> paths;
+        paths.push_back(HAL_LIBRARY_PATH_ODM);
+        if (isHostHal) {
+            if (hostHalPriority) {
+                paths.push_back(HAL_LIBRARY_PATH_VENDOR_EXTRA);
+                paths.push_back(HAL_LIBRARY_PATH_VENDOR);
+            } else {
+                paths.push_back(HAL_LIBRARY_PATH_VENDOR);
+                paths.push_back(HAL_LIBRARY_PATH_VENDOR_EXTRA);
+            }
+        } else {
+            paths.push_back(HAL_LIBRARY_PATH_VENDOR);
+        }
+        paths.push_back(halLibPathVndkSp);
 #ifndef __ANDROID_VNDK__
-            HAL_LIBRARY_PATH_SYSTEM,
+            paths.push_back(HAL_LIBRARY_PATH_SYSTEM);
 #endif
-        };
 
 #ifdef LIBHIDL_TARGET_DEBUGGABLE
         const char* env = std::getenv("TREBLE_TESTING_OVERRIDE");
@@ -751,7 +787,14 @@ sp<::android::hidl::base::V1_0::IBase> getRawServiceInternal(const std::string&
     if (kIsRecovery) {
         transport = Transport::PASSTHROUGH;
     } else {
-        sm = defaultServiceManager1_1();
+        bool hostHalPriority = false;
+        bool isHostHal = false;
+        std::string ifaceName = descriptor.substr(0, descriptor.find("@"));
+
+        if (doesSupportHostBinder()) {
+            isHostHal = isHostHalAllowed(ifaceName, &hostHalPriority);
+        }
+        sm = defaultServiceManager1_1(hostHalPriority);
         if (sm == nullptr) {
             ALOGE("getService: defaultServiceManager() is null");
             return nullptr;
@@ -764,17 +807,16 @@ sp<::android::hidl::base::V1_0::IBase> getRawServiceInternal(const std::string&
                   transportRet.description().c_str());
             return nullptr;
         }
-        if (doesSupportHostBinder())
-        {
+        if (isHostHal) {
             if (transportRet == Transport::EMPTY) {
-                sm = defaultServiceManager1_1(/*useHostHwBinder*/true);
+                sm = defaultServiceManager1_1(!hostHalPriority);
                 if (sm == nullptr) {
-                    ALOGE("getService: defaultServiceManager() of host is null");
+                    ALOGE("getService: defaultServiceManager() is null");
                     return nullptr;
                 }
                 transportRet = sm->getTransport(descriptor, instance);
                 if (!transportRet.isOk()) {
-                    ALOGE("getService: defaultServiceManager()->getTransport of host returns %s",
+                    ALOGE("getService: defaultServiceManager()->getTransport returns %s",
                         transportRet.description().c_str());
                     return nullptr;
                 }
