From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Fri, 26 Nov 2021 15:52:26 -0500
Subject: [PATCH] Add some x86 specific overrides

---
 BoardConfig.mk | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/BoardConfig.mk b/BoardConfig.mk
index d18b8c2..c3205fe 100644
--- a/BoardConfig.mk
+++ b/BoardConfig.mk
@@ -60,6 +60,17 @@ BOARD_SYSTEMIMAGE_PARTITION_SIZE := 3547483648
 BOARD_VENDORIMAGE_PARTITION_SIZE := 1547483648
 BOARD_VENDORIMAGE_FILE_SYSTEM_TYPE := ext4
 
+ifeq ($(filter x86 x86_64,$(TARGET_CPU_ABI)),)
+
+BUILD_BROKEN_USES_BUILD_HOST_EXECUTABLE := true
+BUILD_BROKEN_USES_BUILD_HOST_STATIC_LIBRARY := true
+BUILD_BROKEN_ELF_PREBUILT_PRODUCT_COPY_FILES := true
+
+BUILD_BROKEN_DUP_RULES := true
+RELAX_USES_LIBRARY_CHECK := true
+
+endif
+
 # Disable scudo
 MALLOC_SVELTE := true
 
