From 8c85426287cd525689da0a9d998da278690ed730 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Sun, 1 Mar 2026 03:35:04 +0800
Subject: [PATCH 1/2] Android.bp: Include libva_headers

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Android.bp b/Android.bp
index 3534b33..db494ae 100644
--- a/Android.bp
+++ b/Android.bp
@@ -50,6 +50,7 @@ cc_defaults {
         "libgui",
         "libdrm",
     ],
+    header_libs: ["libva_headers"],
 
     vendor: true,
     enabled: false,
-- 
2.55.0

