From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@lincor.com>
Date: Wed, 1 Apr 2020 17:51:49 -0400
Subject: [PATCH] Add compilation flag -Wno-bool-operation

---
 android/build.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/android/build.mk b/android/build.mk
index 204dd68..a933dcf 100644
--- a/android/build.mk
+++ b/android/build.mk
@@ -67,7 +67,7 @@ LOCAL_CFLAGS += \
 	-Wno-constant-conversion -Wno-incompatible-pointer-types-discards-qualifiers \
 	-Wno-initializer-overrides -Wno-sometimes-uninitialized \
 	-Wno-unneeded-internal-declaration -Wno-unused-function -Wno-absolute-value \
-	-Wno-expansion-to-defined -Wno-string-plus-int \
+	-Wno-expansion-to-defined -Wno-string-plus-int -Wno-bool-operation \
 	-Werror=format-security -Werror=implicit-function-declaration -Werror=missing-prototypes \
 	-Werror=return-type -Werror=vla -Wformat -Wno-maybe-uninitialized -fPIC
 
