From a9e425de94d8efd06c5b9bfb8f43a66324955e06 Mon Sep 17 00:00:00 2001
From: Erfan Abdi <erfangplus@gmail.com>
Date: Fri, 24 Sep 2021 22:54:07 +0330
Subject: [PATCH 3/4] libhardware: Try system hals in vndk lite

---
 hardware.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/hardware.c b/hardware.c
index b4982b5..c6c5688 100644
--- a/hardware.c
+++ b/hardware.c
@@ -78,7 +78,7 @@ static int load(const char *id,
     void *handle = NULL;
     struct hw_module_t *hmi = NULL;
 #ifdef __ANDROID_VNDK__
-    const bool try_system = false;
+    const bool try_system = property_get_bool("ro.vndk.lite", false);
 #else
     const bool try_system = true;
 #endif
@@ -172,6 +172,12 @@ static bool path_in_path(const char *path, const char *in_path) {
 static int hw_module_exists(char *path, size_t path_len, const char *name,
                             const char *subname)
 {
+#ifdef __ANDROID_VNDK__
+    const bool try_system = property_get_bool("ro.vndk.lite", false);
+#else
+    const bool try_system = true;
+#endif
+
     snprintf(path, path_len, "%s/%s.%s.so",
              HAL_LIBRARY_PATH3, name, subname);
     if (path_in_path(path, HAL_LIBRARY_PATH3) && access(path, R_OK) == 0)
@@ -182,12 +188,10 @@ static int hw_module_exists(char *path, size_t path_len, const char *name,
     if (path_in_path(path, HAL_LIBRARY_PATH2) && access(path, R_OK) == 0)
         return 0;
 
-#ifndef __ANDROID_VNDK__
     snprintf(path, path_len, "%s/%s.%s.so",
              HAL_LIBRARY_PATH1, name, subname);
-    if (path_in_path(path, HAL_LIBRARY_PATH1) && access(path, R_OK) == 0)
+    if (try_system && path_in_path(path, HAL_LIBRARY_PATH1) && access(path, R_OK) == 0)
         return 0;
-#endif
 
     return -ENOENT;
 }
-- 
2.30.2

